lms/djangoapps/certificates/tests/test_webview_views.py

```diff
@@ -615,3 +619,35 @@ def test_certificate_custom_template_with_course_mode(self):
             response = self.client.get(test_url)
             self.assertEqual(response.status_code, 200)
             self.assertContains(response, 'mode: {}'.format(mode))
+
+    @ddt.data(True, False)
+    def test_certificate_custom_template_with_unicode_data(self, custom_certs_enabled):
+        """
+        Tests custom template renders properly with unicode data.
+        """
+        mode = 'honor'
+        self._add_course_certificates(count=1, signatory_count=2)
+        self._create_custom_template(mode=mode)
+        with patch.dict("django.conf.settings.FEATURES", {
+            "CERTIFICATES_HTML_VIEW": True,
+            "CUSTOM_CERTIFICATE_TEMPLATES_ENABLED": custom_certs_enabled
+        }):
+            test_url = get_certificate_url(
+                user_id=self.user.id,
+                course_id=unicode(self.course.id)
+            )
+            with patch.dict("django.conf.settings.SOCIAL_SHARING_SETTINGS", {
+                "CERTIFICATE_TWITTER": True,
+                "CERTIFICATE_TWITTER_TEXT": u"nền tảng học tập"
```

Comment:
the issue was with `twitter_url` computations. We make `twitter_url` by combining ascii and non ascii characters in case user has selected a non English language and that was causing the templates to break.